keps/sig-storage/3314-csi-volume-snapshot-delta/README.md

```diff
@@ -0,0 +1,1484 @@
+  "metadata": {
+    "name": "test-delta",
+    "namespace": "default",
+    "creationTimestamp": null
+  },
+  "spec": {
+    "fromVolumeSnapshotName": "vol-snap-base",
+    "toVolumeSnapshotName": "vol-snap-target",
+    "limit": 256,
+    "mode": "block",
+    "offset_bytes": 0
+  },
+  "status": {
+    "limit": 256,
+    "offset_bytes": 0,
+    "continue": 3,
+    "changedBlockDeltas": [
+      {
+        "offset_bytes": 0,
+        "blockSizeBytes": 524288,
+        "dataToken": {
+          "token": "ieEEQ9Bj7E6XR",
+          "issuanceTime": "2022-07-13T03:19:30Z",
+          "ttl": "3h0m0s"
+        }
+      },
+      {
+        "offset_bytes": 524288,
+        "blockSizeBytes": 524288,
+        "dataToken": {
+          "token": "widvSdPYZCyLB",
+          "issuanceTime": "2022-07-13T03:19:30Z",
+          "ttl": "3h0m0s"
+        }
+      },
+      {
+        "offset_bytes": 1048576,
+        "blockSizeBytes": 524288,
+        "dataToken": {
+          "token": "VtSebH83xYzvB",
+          "issuanceTime": "2022-07-13T03:19:30Z",
+          "ttl": "3h0m0s"
+        }
+      }
+    ]
+  }
+}
+```
+
+Any pagination parameters from the storage provider needed to fetch additional
+data will also be included in the response payload to the user. The user will be
+responsible for coordinating subsequent paginated requests, including managing
+the pagination session including recovery from interruption and partition.
+
+### High Availability Mode
+
+To ensure high availability, CSI CBT can be scaled to run multiple replicas of
+the `cbt-aggapi` aggregated API server.
+
+In setup where there may be multiple replicas of the CSI driver, an active/
+passive leader election process will be used to elect a single leader instance
+of the `cbt-http` sidecar, while idling other non-leader instances. Non-leader
+instances will voluntarily fail their readiness probe to remove themselves from
+the `Service`'s request path.
+
+### API Specification
+
+#### VolumeSnapshotDelta Resource
+
+The section describes the specification of `VolumeSnapshotDelta` resource:
+
+```go
+// VolumeSnapshotDelta represents a VolumeSnapshotDelta resource.
+type VolumeSnapshotDelta struct {
+  metav1.TypeMeta `json:",inline"`
+
+  // Standard object's metadata.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // spec defines the desired characteristics of a snapshot delta requested by a user.
+  // Required.
+  Spec VolumeSnapshotDeltaSpec `json:"spec"`
+
+  // status represents the current information of a snapshot delta.
+  // +optional
+  Status VolumeSnapshotDeltaStatus `json:"status,omitempty"`
+}
+
+// VolumeSnapshotDeltaSpec is the spec of a VolumeSnapshotDelta resource.
+type VolumeSnapshotDeltaSpec struct {
+  // The name of the base CSI volume snapshot to use for comparison.
+  // If not specified, return all changed blocks.
```

Comment:
Return all changed blocks since when? Volume creation? It should be specified there.